internal/remote/user_queue.go

```diff
@@ -73,6 +74,19 @@ func (uoq *userOpQueue) tryGetNextOp() operation {
 	return nil
 }
 
+func (user *User) newContextWithIMAPID(id ConnMetadataID) context.Context {
```

Comment:
I'd prefer if the function signature here matched that of `imap.NewContextWithIMAPID(ctx, x)`, that is, it takes in a parent context and returns a child context.

You could even pass in the parent context (which would just be `context.Background()` as an argument to `user.process(...)`.

Reply:
Fixed.